File: codility/CommonPrimeDivisors.js
function solution(A, B) {
    // Implement your solution here3.50 30 350
    let counter = 0;
    for (let i = 0; i < A.length; i++) {
        const a = A[i],
            b = B[i];
        if (a === b) {
            counter++;
            continue;
        }
        const gcd = getgcd(a, b);
        if (gcd === 1) {
            continue;
        }
        const c = a / gcd;
        const d = b / gcd;
        if (!hasOtherPrime(c, gcd) && !hasOtherPrime(d, gcd)) {
            counter++;
        }
    }
    return counter;
    //判断n是否有gcd包含约数之外的约数10,8
    function hasOtherPrime(n, gcd) {
        const gcdN = getgcd(n, gcd);
        if (gcdN === 1) {
            return n !== 1;
        }
        return hasOtherPrime(n / gcdN, gcd);
    }
    function getgcd(a, b) {
        if (a < b) {
            return getgcd(b, a);
        }
        if (a % b === 0) {
            return b;
        }
        return getgcd(b, a % b);
    }
}

File: codility/python/CommonPrimeDivisors.py
def solution(A, B):
    # Implement your solution here
    n = len(A)
    count = 0
    for i in range(n):
        a = A[i]
        b = B[i]
        if a == b:
            count += 1
            continue
        gcd = get_gcd(a, b)
        ad = int(a / gcd)
        bd = int(b / gcd)
        if has_other_prime(ad, gcd) == False and has_other_prime(bd, gcd) == False:
            count += 1
            continue
    return count

def get_gcd(a, b):
    if a < b:
        return get_gcd(b, a)
    if a % b == 0:
        return b
    return get_gcd(b, a % b)

def has_other_prime(n, gcd):
    g = get_gcd(n, gcd)
    if g == 1:
        return n != 1
    return has_other_prime(int(n / g), gcd)